File: Data Structures & Algorithms/duplicate-integer/submission-4.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        // Hashset solution in Java with enhanced for loop and Set

        Set<Integer> numsHash = new HashSet<Integer>();
        for (int num : nums) {
            if (!numsHash.add(num)){
                return true;
            }
        } return false;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 0
        nums_set = set(nums)

        for num in nums_set:
            
            length = 1

            if num-1 not in nums_set:
                while num+1 in nums_set:
                    length+=1
                    num+=1
            
            longest = max(longest, length)
        
        return longest